juicer/spark/spark_minion.py

```diff
@@ -935,14 +935,10 @@ def terminate(self):
         minion. In this case, we stop and release any allocated resource
         (spark_session) and kill the subprocess managed in here.
         """
-        if self.spark_session and multiprocessing.current_process().name == 'main':
+        if self.spark_session and multiprocessing.current_process().name == 'MainProcess':
             try:
-                # sc = self.spark_session.sparkContext
```

Comment:
This old code was never executed because the right name is "MainProcess". Besides that, the instantiation of variable sc was commented, resulting in an error. 

In Spark, when we call stop(), it only stops the section, the java process continues to exist. Because of that, you were probably trying to run `shutdown_callback_server` to force it, right?. Because of we use multiprocessing, this command is not necessary, the java process will be deleted at end.  Besides that, this old approach will not work if you try to create a new session, you will need to reload some modules.